src/component/Text.js

```diff
@@ -0,0 +1,162 @@
+import reduceCSSCalc from 'reduce-css-calc';
+import { PRESENTATION_ATTRIBUTES } from '../util/ReactUtils';
+import _ from 'lodash';
+
+class Text extends Component {
+
+  static propTypes = {
+    ...PRESENTATION_ATTRIBUTES,
+    fit: PropTypes.bool,
+    angle: PropTypes.number,
+    textAnchor: PropTypes.oneOf(['start', 'middle', 'end', 'inherit']),
+    verticalAnchor: PropTypes.oneOf(['start', 'middle', 'end']),
+  };
+
+  static defaultProps = {
+    x: 0,
+    y: 0,
+    lineHeight: '1em',
+    capHeight: '0.71em', // Magic number from d3
+    fit: false,
+    textAnchor: 'start',
+    verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
+  };
+
+  state = {
+    wordsByLines: [],
+  };
+
+  componentWillMount() {
+    this.updateWordsByLines(this.props, true);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const calculateWordWidths = (
+      this.props.children !== nextProps.children ||
+      this.props.style !== nextProps.style
+    );
+    this.updateWordsByLines(nextProps, calculateWordWidths);
+  }
+
+  updateWordsByLines(props, calculateWordWidths) {
+    // Only perform calculations if using features that require them (multiline, fit)
+    if (props.width || props.fit) {
+      if (calculateWordWidths) {
+        const { wordsWithComputedWidth, spaceWidth } = this.calculateWordWidths(props);
+        this.wordsWithComputedWidth = wordsWithComputedWidth;
+        this.spaceWidth = spaceWidth;
+      }
+
+      const wordsByLines = this.calculateWordsByLines(
+        this.wordsWithComputedWidth,
+        this.spaceWidth,
+        props.width
+      );
+      this.setState({ wordsByLines });
+    } else {
+      const words = props.children ? props.children.toString().split(/\s+/) : [];
+      this.setState({ wordsByLines: [{ words }] });
+    }
+  }
+
+  calculateWordWidths(props) {
+    // Calculate length of each word to be used to determine number of words per line
+    var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    var text = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+    _.assign(text.style, props.style);
+    svg.appendChild(text);
+    document.body.appendChild(svg);
+
+    const words = props.children ? props.children.toString().split(/\s+/) : [];
+    const wordsWithComputedWidth = words.map(word => {
+      text.textContent = word;
+      return { word, width: text.getComputedTextLength() };
+    });
+
+    text.textContent = '\u00A0'; // Unicode space
+    const spaceWidth = text.getComputedTextLength();
+
+    document.body.removeChild(svg);
+
+    return { wordsWithComputedWidth, spaceWidth };
+  }
+
+  calculateWordsByLines(wordsWithComputedWidth, spaceWidth, lineWidth) {
+    const { fit } = this.props;
+    return wordsWithComputedWidth.reduce((result, { word, width }) => {
+      const currentLine = result[result.length - 1] || { words: [], width: 0 };
+
+      if (currentLine.words.length === 0) {
```

Comment:
It seems duplicated here. Maybe these lines can be:

```js
const currentLine = result[result.length - 1];

if (currentLine && (lineWidth == null || fit || (currentLine.width + width + spaceWidth) < lineWidth)) {
   currentLine.words.push(word);
   currentLine.width += width + spaceWidth;
} else {
   const newLine = { words: [word], width };
   result.push(newLine);
}
```

But your implementation is clearer...